src/main/java/org/opensearch/security/ssl/util/SSLConfigConstants.java

```diff
@@ -32,14 +32,18 @@ public final class SSLConfigConstants {
     public static final String SECURITY_SSL_HTTP_KEYSTORE_ALIAS = "plugins.security.ssl.http.keystore_alias";
     public static final String SECURITY_SSL_HTTP_KEYSTORE_FILEPATH = "plugins.security.ssl.http.keystore_filepath";
     public static final String SECURITY_SSL_HTTP_PEMKEY_FILEPATH = "plugins.security.ssl.http.pemkey_filepath";
+    @Deprecated // in favor of the _secure variant
```

Comment:
What do you think about deleting these constants and depending on something defined in the SSLSettings?

Reply:
I would be for deleting them entirely as you have to extract them from the settings using the SecureSetting.get call (which caught me out at first). But thought that marking them deprecated in the short term might be a better approach.

I also thought that the secure variants should be private too, to stop other devs accidentally calling setting.get(xxx_SECURE), which does not do what you would expect it to do (as noted above)

Happy to accept guidance here on just removing them instead.

Reply:
One of the reasons I like the enum approach where you'd get a value with `final String password = SSLSettings.HTTP_KEYSTORE_PASSWORD.getFromSetting(settings);`  is that internally it could attempt the secure named value first, then attempt the fallback to insecure, and finally do whatever we do when the setting isn't set.  This hides the usage in the product that there is more than one way to set this value.

Reply:
This should be resolved in the latest push,  but there are other issues we can discuss elsewhere